integration_tests/src/main/python/arithmetic_ops_test.py

```diff
@@ -500,3 +501,32 @@ def test_greatest(data_gen):
             lambda spark : gen_df(spark, gen).select(
                 f.greatest(*command_args)), conf=allow_negative_scale_of_decimal_conf)
 
+def _assert_div_by_zero(session_func, test_div_func, conf):
+    with pytest.raises(Py4JJavaError) as py4jError:
+        session_func(test_div_func, conf)
+    assert 'java.lang.ArithmeticException: divide by zero' in str(py4jError.value.java_exception)
+
+@pytest.mark.parametrize('ansi_mode', ['nonAnsi', 'ansi'])
+@pytest.mark.parametrize('exp_type', ['bothLiterals', 'justZeroLiteral', 'noLiterals'])
+def test_div_by_zero(ansi_mode, exp_type):
+    if ansi_mode == 'ansi':
+        if is_before_spark_310() or exp_type != 'bothLiterals':
+            pytest.xfail('https://github.com/apache/spark/pull/29882')
+
+    ansi_conf = {'spark.sql.ansi.enabled': ansi_mode == 'ansi'}
+    data_gen = lambda spark: two_col_df(spark, IntegerGen(), IntegerGen(min_val=0, max_val=0), length=1)
+
+    if exp_type == 'bothLiterals':
+        div_by_zero_func = lambda spark: data_gen(spark).select(f.lit(1) / f.lit(0))
```

Comment:
I don't think we need to test 'bothLiterals'.  In Spark when both parts of an operation are literal values it runs that code as a part of optimizations and will replace it with the a new literal that is the result of the operation before our code even has a chance to run.  So you are only testing that Spark is behaving correctly, not our code.

Reply:
yes that documents that we are covered with the checks in constant folding

Reply:
I am fine with it, but it is inconsistent with all of the other tests that we have done.